### ISSUE TITLE
Training on an RTX 5090 is painfully slow, and I’m unable to configure the "(Optional) Manually configure Accelerate" option.

### ISSUE THREAD

**Issue by Synergyart** [bmaltais/kohya_ss#3332]
Hello! I’m hoping someone can help me. I just reinstalled Windows 11 on my PC and set up Kohya_ss from scratch. When I tested training a LoRA with Illustrious XL, this happened:

1. I could not configure Accelerate at all. I typed “5” and pressed Enter, but nothing happened.

<img width="1149" height="668" alt="Image" src="https://github.com/user-attachments/assets/feb8faef-da3c-4628-8da7-1d6514dbf52e" />

2. Training a 2 image LoRA now takes 4.78 s/it. I’m using batch size 1 with 40 repeats.
<img width="1119" height="621" alt="Image" src="https://github.com/user-attachments/assets/238a65fd-fe09-4bd6-90c4-7c8ad0a81ed1" />

Aside from speed, the LoRA comes out fine, no other issues. Before reinstalling Windows 11 and Kohya_ss, I was getting 1–2 s/it.

Setup:
- GPU: RTX 5090  
- Windows 11  
- Kohya_ss branch: “dev”  
- CUDA version (tried 12.9 and nothing):
  nvcc: NVIDIA (R) Cuda compiler driver  
  Copyright (c) 2005-2025 NVIDIA Corporation  
  Built on Wed Jan 15 19:38:46 PST 2025  
  Cuda compilation tools, release 12.8, V12.8.61  
  Build cuda_12.8.r12.8/compiler.35404655_0  
- PyTorch: the version included with Kohya_ss “dev”

Here is the JSON I use to train (it worked fine on my RTX 3090):

[Illus.json](https://github.com/user-attachments/files/21134195/Illus.json)

**Comment by b-fission** [comment]
You can turn off "Gradient checkpointing" since that'll cut your speed in half. That option helps reduce vram usage, but that's not needed since you're on a 5090.

**Comment by Synergyart** [comment]
> You can turn off "Gradient checkpointing" since that'll cut your speed in half. That option helps reduce vram usage, but that's not needed since you're on a 5090.


Thank you. I disabled "Gradient checkpointing," which improved the time quite a bit. However, I'm still getting 4 s/it, and I haven't been able to configure Accelerate,  maybe that's affecting performance too?

<img width="1222" height="722" alt="Image" src="https://github.com/user-attachments/assets/175d0012-7e42-452d-b6c9-eb98a66e2ed1" />

new .json with Gradient checkpointing off:

[Illus.json](https://github.com/user-attachments/files/21135075/Illus.json)

**Comment by b-fission** [comment]
I was expecting a much bigger improvement. Going from 4.7 s/it --> 4.0 s/it is mild.
The config you posted is getting 1.1 s/it on my 4090.

Does the training speed change when you set the cross-attention to "sdpa" instead of xformers?
Or try a different optimizer like AdamW?


-----

Doing "Configure accelerate" isn't super necessary. All it does is generate this little config file. Most of these settings can be overridden in the kohya gui's "Accelerate launch" category.
```yaml
{
  "compute_environment": "LOCAL_MACHINE",
  "distributed_type": "NO",
  "downcast_bf16": false,
  "machine_rank": 0,
  "main_training_function": "main",
  "mixed_precision": "no",
  "num_machines": 1,
  "num_processes": 2,
  "rdzv_backend": "static",
  "same_network": false,
  "tpu_use_cluster": false,
  "tpu_use_sudo": false,
  "use_cpu": false
}
```

**Comment by Synergyart** [comment]
> I was expecting a much bigger improvement. Going from 4.7 s/it --> 4.0 s/it is mild. The config you posted is getting 1.1 s/it on my 4090.
> 
> Does the training speed change when you set the cross-attention to "sdpa" instead of xformers? Or try a different optimizer like AdamW?
> 
> Doing "Configure accelerate" isn't super necessary. All it does is generate this little config file. Most of these settings can be overridden in the kohya gui's "Accelerate launch" category.
> 
> {
>   "compute_environment": "LOCAL_MACHINE",
>   "distributed_type": "NO",
>   "downcast_bf16": false,
>   "machine_rank": 0,
>   "main_training_function": "main",
>   "mixed_precision": "no",
>   "num_machines": 1,
>   "num_processes": 2,
>   "rdzv_backend": "static",
>   "same_network": false,
>   "tpu_use_cluster": false,
>   "tpu_use_sudo": false,
>   "use_cpu": false
> }

### Previous optimizer

Here are the results with "sdpa". It lowers the time to 3.91s/it:

<img width="1122" height="1583" alt="Image" src="https://github.com/user-attachments/assets/1aff5147-7081-4111-a054-4dd063555855" />

Below is the MSI Afterburner configuration, followed by the readings from HWiNFO and Task Manager:

<img width="1491" height="828" alt="Image" src="https://github.com/user-attachments/assets/6e2e8354-f93c-47d6-a44b-54cba72b0aa4" />

<img width="980" height="1437" alt="Image" src="https://github.com/user-attachments/assets/c0360cd5-0017-477c-8018-18483c587fa8" />

The JSON settings are the same as before, without gradient checkpointing.

---
### AdamW


I tested AdamW using some of the settings from this video: https://www.youtube.com/watch?v=y2J7EZUk_a0  
Below is the JSON file I used (the previous version failed due to parameter errors):  

[TEST Illus AdamW.json](https://github.com/user-attachments/files/21136710/TEST.Illus.AdamW.json)

The timing results were much better, whether I used xFormers or sdpa. sdpa remained the faster option.

sdpa:

<img width="1115" height="688" alt="Image" src="https://github.com/user-attachments/assets/1e612cef-88ee-40fe-8a22-83ad7987d30c" />

xFormers:

<img width="1113" height="686" alt="Image" src="https://github.com/user-attachments/assets/1489ca11-7bd4-4791-ac12-aa894f25b652" />

It’s still not down to 1.1 s/it like your GPU, but it’s a significant improvement! WOW! I need to stay away from Prodigy it seems.

**Comment by Syscore64** [comment]
I Gain at SDXL LoRA Training 3,45s/it
**Batch Size 4**
with **gradient accumulation steps: 2**
Alpha 16
Dim 32
Size: 1024x1024

MSI Ventus 5090
2805mhz Core
16801Mhz VRAM
890mV Core Voltage

Whats your CUDA / Xformers / PyTorch version?

<img width="1902" height="402" alt="Image" src="https://github.com/user-attachments/assets/c7b2131c-b936-464c-a28b-144d8be6077b" />

![Image](https://github.com/user-attachments/assets/3fa185a8-ab77-4696-b622-aa38448afc53)

**Comment by Synergyart** [comment]
> I Gain at SDXL LoRA Training 3,45s/it **Batch Size 4** with **gradient accumulation steps: 2** Alpha 16 Dim 32 Size: 1024x1024
> 
> MSI Ventus 5090 2805mhz Core 16801Mhz VRAM 890mV Core Voltage
> 
> Whats your CUDA / Xformers / PyTorch version?
> 
> <img alt="Image" width="1902" height="402" src="https://private-user-images.githubusercontent.com/38880076/466184115-c7b2131c-b936-464c-a28b-144d8be6077b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI1MzAxMzEsIm5iZiI6MTc1MjUyOTgzMSwicGF0aCI6Ii8zODg4MDA3Ni80NjYxODQxMTUtYzdiMjEzMWMtYjkzNi00NjRjLWEyOGItMTQ0ZDhiZTYwNzdiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE0VDIxNTAzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3ZTM3NjA3ZWIyMDdiYTM5M2JiMzBlNmE4OTBiMDA0Njc0MTAzNjZmMTczNmZmMjg4OTY0ODgwZjA4NzA4YmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.IxFxVcdFcREP66ToXrlnELswBbdQo5qs0Hyy2noOIeQ">
> ![Image](https://github.com/user-attachments/assets/3fa185a8-ab77-4696-b622-aa38448afc53)

Try using my **.json**, it should work on any SDXL without any changes, then see what to change:

[Illus.AdamW.json](https://github.com/user-attachments/files/21222844/Illus.AdamW.json)



If you have **Gradient checkpointing** on, disable it. I never used gradient accumulation so i can't help there. Also, see if sysmem fallback policy is off and what your vram is doing.

Other:
I use the Xformers and PyTorch from the gui-uv.bat
My cuda: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2025 NVIDIA Corporation
Built on Tue_May_27_02:24:01_Pacific_Daylight_Time_2025
Cuda compilation tools, release 12.9, V12.9.86
Build cuda_12.9.r12.9/compiler.36037853_0
cuDNN: 9.10.2

**Comment by Syscore64** [comment]
> > I Gain at SDXL LoRA Training 3,45s/it **Batch Size 4** with **gradient accumulation steps: 2** Alpha 16 Dim 32 Size: 1024x1024
> > MSI Ventus 5090 2805mhz Core 16801Mhz VRAM 890mV Core Voltage
> > Whats your CUDA / Xformers / PyTorch version?
> > <img alt="Image" width="1902" height="402" src="https://private-user-images.githubusercontent.com/38880076/466184115-c7b2131c-b936-464c-a28b-144d8be6077b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI1MzAxMzEsIm5iZiI6MTc1MjUyOTgzMSwicGF0aCI6Ii8zODg4MDA3Ni80NjYxODQxMTUtYzdiMjEzMWMtYjkzNi00NjRjLWEyOGItMTQ0ZDhiZTYwNzdiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE0VDIxNTAzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3ZTM3NjA3ZWIyMDdiYTM5M2JiMzBlNmE4OTBiMDA0Njc0MTAzNjZmMTczNmZmMjg4OTY0ODgwZjA4NzA4YmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.IxFxVcdFcREP66ToXrlnELswBbdQo5qs0Hyy2noOIeQ">
> > ![Image](https://github.com/user-attachments/assets/3fa185a8-ab77-4696-b622-aa38448afc53)
> 
> Try using my **.json**, it should work on any SDXL without any changes, then see what to change:
> 
> [Illus.AdamW.json](https://github.com/user-attachments/files/21222844/Illus.AdamW.json)
> 
> If you have **Gradient checkpointing** on, disable it. I never used gradient accumulation so i can't help there. Also, see if sysmem fallback policy is off and what your vram is doing.
> 
> Other: I use the Xformers and PyTorch from the gui-uv.bat My cuda: NVIDIA (R) Cuda compiler driver Copyright (c) 2005-2025 NVIDIA Corporation Built on Tue_May_27_02:24:01_Pacific_Daylight_Time_2025 Cuda compilation tools, release 12.9, V12.9.86 Build cuda_12.9.r12.9/compiler.36037853_0 cuDNN: 9.10.2

You misunderstand, with batch 1 and gradient checkpointing 1 I have 1.24 it/s with your json 1.08it/s, should be much faster with your 5090.
Unfortunately I am not familiar with the UV version.
